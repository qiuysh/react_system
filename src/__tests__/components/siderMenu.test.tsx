/** @format */
import * as React from "react";
import {
  render,
  cleanup,
  fireEvent,
  RenderResult,
  logRoles,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import SiderMenu from "@components/siderMenu";

const defaultProps = {
  collapsed: false,
  history: {
    location: {
      pathname: "/",
    },
  },
  menuList: [
    {
      id: 1,
      pid: 0,
      name: "仪表盘",
      type: 1,
      icon: "icondashboard",
      url: "/dashboard",
      code: "dashboard",
      desc: null,
      create_time: "2021-03-10T10:47:52.000Z",
      modified_time: null,
    },
  ],
};

describe("test siderMenu", () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = render(<SiderMenu {...defaultProps} />);
  });

  afterEach(cleanup);

  it("test siderMenu -- snapshot", () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it("test siderMenu -- click event", () => {
    const menu = wrapper.getByTestId("test-sider-nav");
    const handleClick = jest.fn();
    fireEvent.click(menu);
    expect(handleClick).toHaveBeenCalled();
  });
});
