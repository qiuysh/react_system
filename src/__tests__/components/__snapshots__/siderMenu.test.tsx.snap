// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test siderMenu test siderMenu -- snapshot 1`] = `
<DocumentFragment>
  <aside
    class="ant-layout-sider ant-layout-sider-dark yux-sider"
    style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <div
        class="logo"
      >
        Idev Work
      </div>
      <ul
        class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark"
        data-menu-list="true"
        data-testid="test-sider-nav"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
          data-menu-id="rc-menu-uuid-test-dashboard"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span>
              仪表盘
            </span>
          </span>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
  </aside>
</DocumentFragment>
`;
